development/build/index.js

```diff
@@ -164,89 +166,110 @@ function defineAndRunBuildTasks() {
   );
 
   // special build for minimal CI testing
-  createTask('styles', styleTasks.prod);
+  createTask(TASKS.styles, styleTasks.prod);
 
   // Finally, start the build process by running the entry task.
   runTask(entryTask, { skipStats });
 }
 
 function parseArgv() {
-  const NamedArgs = {
-    ApplyLavaMoat: 'apply-lavamoat',
-    BuildType: 'build-type',
-    BuildVersion: 'build-version',
-    LintFenceFiles: 'lint-fence-files',
-    Lockdown: 'lockdown',
-    PolicyOnly: 'policy-only',
-    SkipStats: 'skip-stats',
-  };
-
-  const argv = minimist(process.argv.slice(2), {
-    boolean: [
-      NamedArgs.ApplyLavaMoat,
-      NamedArgs.LintFenceFiles,
-      NamedArgs.Lockdown,
-      NamedArgs.PolicyOnly,
-      NamedArgs.SkipStats,
-    ],
-    string: [NamedArgs.BuildType, NamedArgs.BuildVersion],
-    default: {
-      [NamedArgs.ApplyLavaMoat]: true,
-      [NamedArgs.BuildType]: BuildType.main,
-      [NamedArgs.BuildVersion]: '0',
-      [NamedArgs.LintFenceFiles]: true,
-      [NamedArgs.Lockdown]: true,
-      [NamedArgs.PolicyOnly]: false,
-      [NamedArgs.SkipStats]: false,
-    },
-  });
+  const { argv } = yargs(hideBin(process.argv))
+    .usage('$0 <task> [options]', 'Build the MetaMask extension.', (_yargs) =>
+      _yargs
+        .positional('task', {
+          description: `The task to run. There are a number of main tasks, each of which calls other tasks internally. The main tasks are:
 
-  if (argv._.length !== 1) {
-    throw new Error(
-      `Metamask build: Expected a single positional argument, but received "${argv._.length}" arguments.`,
-    );
-  }
+prod: Create an optimized build for a production environment.
 
-  const entryTask = argv._[0];
-  if (!entryTask) {
-    throw new Error('MetaMask build: No entry task specified.');
-  }
+dev: Create an unoptimized, live-reloading build for local development.
 
-  const buildType = argv[NamedArgs.BuildType];
-  if (!(buildType in BuildType)) {
-    throw new Error(`MetaMask build: Invalid build type: "${buildType}"`);
-  }
+test: Create an optimized build for running e2e tests.
 
-  const rawBuildVersion = argv[NamedArgs.BuildVersion];
-  const buildVersion = Number.parseInt(rawBuildVersion, 10);
-  if (rawBuildVersion.match(/^\d+$/u) === null || Number.isNaN(buildVersion)) {
-    throw new Error(
-      `MetaMask build: Invalid build version: "${rawBuildVersion}"`,
-    );
-  }
+testDev: Create an unoptimized, live-reloading build for debugging e2e tests.`,
+          type: 'string',
```

Comment:
I used `string` here rather than `choices` because there were too many tasks, it was overwhelming to see them all listed out. Most of them aren't meant to be called directly anyway. So instead the description lists the four top-level tasks.